framework/src/main/java/org/apache/drill/test/framework/DrillTestJdbc.java

```diff
@@ -95,8 +95,16 @@ public void run() {
       }
 
       queries = Utils.getSqlStatements(modeler.queryFilename);
-      mainQueryIndex = queries.length / 2; // Currently, the main query must be in the middle of the list of queries
-
+      for (int i = 0; i < queries.length && mainQueryIndex == 0; i++) {
+    	if (queries[i].startsWith("@")) {
```

Comment:
It might be worthwhile to check for `--@` instead of just `@`
This allows the file to be consumed by other SQL Engines too, since a leading double hyphen is treated by most SQL parsers as comments. 
You could then expand the annotations to do more advanced things, on similar lines of how test frameworks like JUnit, etc. interpret java code as elements of test - e.g. setup, test, cleanup.